docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
+### 4.1. Product Scope
+
+NUSMaze is targeted at NUS engineering freshman, to help new students find their way to their destination blocks.
+
+The engineering block is extremely huge, and the layout of the blocks may be confusing for new students. To reduce the time wasted on navigating the numerous blocks in Engineering, NUSMaze will provide the shortest route available for students to take.
+
+### 4.2. User Stories
+
+|Version| As a ... | I want to ... | So that I ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|new user|see usage instructions|can refer to them when I forget how to use the application|
+|v1.0|user|have a clear path to my destination|will not get lost|
+|v1.0|user|be able to pin a note to certain locations as a reminder|do not forget|
+|v1.0|user|keep track of my search history|don't have to repeatedly search for the same route.|
+|v1.0|user|have a clear interface in which I can enter my commands|can have a good user experience|
+|v2.0|user|find the nearest eatery|do not have to starve for longer than necessary|
+|v2.0|user|have a list of favorite locations|can access directions to them quickly|
+|v2.0|user|have my list of favourites and history stored|can access it every time I start the app|
+|v2.0|user|be able to set custom aliases to blocks|can access the blocks more conveniently|
+|v2.0|user|be able to store my routing for my daily activities|can access it easily|
+
+### 4.3. Use Cases
+
+### 4.4. Non-Functional Requirements
+
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+2. Should be able to hold up to 1000 history, notes, favourites and block alias entries without a noticeable sluggishness in performance for typical usage.
 
-## Non-Functional Requirements
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
-{Give non-functional requirements}
+### 4.5. Glossary
 
-## Glossary
+* *CLI* - Command Line Interface
 
-* *glossary item* - Definition
+---------------------------------------------------------------------------------------------
 
-## Instructions for manual testing
+## *5. Appendix: Instructions for manual testing*
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### 5.1 Launch and shutdown
+1. Initial launch  
+    1.1. Download the latest jar file from [here](https://github.com/AY2021S2-CS2113T-T09-2/tp/releases/tag/v1.0)
+   and copy it into an empty folder.  
+    1.2. Open the terminal/powershell console and navigate to the folder in which the `.jar` file was saved.  
+   Expected: Greeting message of NUSMaze  to be shown
```

Comment:
Maybe can show the expected output in a screenshot as what you did in 1.2 Setting up and getting started